homeassistant/components/linky/config_flow.py

```diff
@@ -27,19 +27,9 @@ class LinkyFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
 
     def __init__(self):
         """Initialize Linky config flow."""
-        self._username = None
-        self._password = None
-        self._timeout = None
-
-    def _configuration_exists(self, username: str) -> bool:
-        """Return True if username exists in configuration."""
-        for entry in self.hass.config_entries.async_entries(DOMAIN):
-            if entry.data[CONF_USERNAME] == username:
-                return True
-        return False
-
-    @callback
-    def _show_setup_form(self, user_input=None, errors=None):
+        pass
+
+    async def _show_setup_form(self, user_input=None, errors=None):
```

Comment:
Why make it async ?

Reply:
It's async in `adguard`, `icloud`, `twentemilieu` and also not a `callback` (but not async) in `elgato` and `wled`.

Don't know exactly which is the best, but can fix other components if needed.

No callback no async
No callback async
Callback no async
Callback async

Getting confused, it's not homogen in the code.

Reply:
It's best to not define coroutines if we don't need to. There's an overhead to awaiting a coroutine compared to running a callback in the event loop.